main/test/ca/uwaterloo/flix/language/phase/TestRedundancy.scala

```diff
@@ -1278,4 +1278,103 @@ class TestRedundancy extends FunSuite with TestUtils {
     val result = compile(input, Options.TestWithLibNix)
     expectError[RedundancyError.RedundantUpcast](result)
   }
+
+  test("TestUnreachableTypeMatchCase.01") {
+    val input =
+      """
+        |def f(): Unit =
+        |    typematch () {
+        |        case _: Int32 => ()
+        |        case _: _ => ()
+        |    }
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[RedundancyError.UnreachableTypeMatchCase](result)
+  }
+
+  test("TestUnreachableTypeMatchCase.02") {
+    val input =
+      """
+        |enum E[_] {
+        |    case E
+        |}
+        |
+        |def f(): Unit =
+        |    typematch E {
+        |        case _: E[Int32] => ()
+        |        case _: _ => ()
+        |    }
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[RedundancyError.UnreachableTypeMatchCase](result)
+  }
+
+  test("TestUnreachableTypeMatchCase.03") {
+    val input =
+      """
+        |enum E[_] {
+        |    case E
+        |}
+        |
+        |def f(_: a): Unit =
+        |    typematch E {
+        |        case _: E[a] => ()
```

Comment:
What's the reasoning here?

Reply:
See above. `a` is bound so it's rigid, acting like the `Int32` in the other example.